### ISSUE TITLE
The training results

### ISSUE THREAD

**Issue by junwenxiong** [fuankarion/MAAS#2]
I have successfully processed ava-activespeaker Dataset and trained MAAS with forward embedding. The following results are obtained:
![image](https://user-images.githubusercontent.com/35716657/156768065-61a16bb3-a8ec-420f-b962-fe02cb52d05f.png)

This result surprised me very much. The effect of the model on the training set is good, but the effect on the validation set is too poor. Why does this happen?

The training parameters I use are as follows:
`python maas_train.py 13 1 3 64 0 3`

**Comment by fuankarion** [comment]
Hi, 
The second parameter (stride) must be set to 3 or 4 ( I don't fully recall the value and is not documented, sorry about the extra experiment). 
But that's not the major issue here, that 0.46 mAP definitely rings a bell and I think that's what happens when you have bad features. Can you double check the training features are meaningful (e.g. not zero, and not the same value across the dataset). Also double check that you are loading load all the features from all the videos, when you start that script you see a console output with all the videos being loaded to RAM (120 train, 33 val ), it should require about 50GB of ram to load the full set.

**Comment by fuankarion** [comment]
Ohh and the best argument for the speaker number is 4. but even with the stride at 1 you should be around 88 mAP. not 44-46

**Comment by woshihaozhu** [comment]
Thank you very much for offering that important information! I guess maybe the process of extracting features happened in some accident. Because, the process of training the STE model was very smooth, and the following is the result of my training.
![image](https://user-images.githubusercontent.com/35716657/156877703-c535b1f9-e4d5-4dcb-b1ee-dcd045fa28df.png)
Are these results right?

**Comment by fuankarion** [comment]
Those results are close but not quite there (should be around 77-78). What size did you chose for the clip of the STE? it should be 11 or 13. Did you change the input size?
Anyway this doesn't explain the poor results on the second stage, have you checked if the features? I think the code allows you to do a forward pass if the paths to the video or audio data are wronf, it fails silently and forward and empty image and empty audio track.

**Comment by woshihaozhu** [comment]
After several days of effort, the model still cannot achieve the performance effect in the paper. First of all, I retrained the STE model, and its performance in the validation set reached 0.79, which on the other hand shows that the split of the data set should be no problem.
![image](https://user-images.githubusercontent.com/35716657/157234176-78a03e6e-c71b-4e04-aa1e-cb967385d1e0.png)
Next, the pre-training model is used to extract multi-modal embedding, and all parameters are provided by the repository without any modification.
![image](https://user-images.githubusercontent.com/35716657/157234896-3aea3d37-eaae-4f42-9b24-92492d6f460f.png)
Finally, we use the obtained embedding to train the MAAS model and set the parameters as 13 4 4 64 3.  The effect on the training set seems to be correct, but the model performance on the validation set is different from what I expected.
![image](https://user-images.githubusercontent.com/35716657/157235264-2d972871-849f-4fc3-bd6c-3a4a1a9c8178.png)

So, I'm a little confused about how this happens. If may, can you share the processed data?

**Comment by fuankarion** [comment]
The numbers for the STE encoder now look right, since the second stage is not improving I think you have an error in your features, could you double check this dictionary: 
https://github.com/fuankarion/MAAS/blob/19b42d72d8c10a85d848464a34e78c3244afb5cb/core/config.py#L58
See if those paths actually exists and contain the new feature set.
If nothing seems wrong, could you share a couple feature files? that 46mAP is almost always a somewhat random prediction made from bad features.

**Comment by woshihaozhu** [comment]
The config file used in training is consistent with that in the repository without any changes. In addition, I have uploaded some feature files of train and val set, you can check out this link. Thank you!
[https://drive.google.com/drive/folders/1KkEwnf4EcIBNego0lFIrPBOtSAzNSW-p?usp=sharing](url)

**Comment by fuankarion** [comment]
Thanks for the files. They look well built, so this must be an error en the data loading or the training, give me some time to check the code and I'll get it to work on the feature files you just sent.

**Comment by fuankarion** [comment]
Sorry but I can't reproduce the bug with the provided data

I simply cloned the repo and pointed to your files and got the following on the fist epoch:
![MAAS](https://user-images.githubusercontent.com/16943391/157446825-c1977dfd-f898-4ce4-b241-b203a2d0b582.png)

The only thing I can think about is that maybe you are pointing to old csv files (obtained with another STE)?
Can you maybe clone this repo to a new destination and run over the very same files you sent me, do you still get bad results?

**Comment by fuankarion** [comment]
@woshihaozhu, Wait a second, the repo didn't clone and it actually used my last local version for the run above. There is one line that is probably causing a mess. I just corrected it. Please pull and tell me if you still have the issues.

**Comment by junwenxiong** [comment]
Thanks for your help. I'm just giving it a try.

**Comment by junwenxiong** [comment]
Hha, the latest code is useful, which is amazing, just changing the len function in the dataset from 20 to 1. 
![image](https://user-images.githubusercontent.com/35716657/157457758-8fd07e3a-f237-499c-a395-0ae2159b7cea.png)

Great, I can finally go ahead. It was a great job and inspired me a lot, and I want to keep improving on your code, and if I succeed, I'll cite this article. Thank you very much.
